runtime/vam/expr/function/flatten.go

```diff
@@ -31,3 +31,25 @@ func (f *flatten) Call(args ...vector.Any) vector.Any {
 	}
 	return builder.Build()
 }
+
+type unflatten struct {
+	fn *samfunc.Unflatten
+}
+
+func newUnflatten(zctx *super.Context) *unflatten {
+	return &unflatten{samfunc.NewUnflatten(zctx)}
+}
+
+func (u *unflatten) Call(args ...vector.Any) vector.Any {
+	vec := vector.Under(args[0])
+	typ := vec.Type()
+	builder := vector.NewDynamicBuilder()
+	var b zcode.Builder
+	for i := range vec.Len() {
+		b.Reset()
```

Comment:
Nit: Should be safe to use Truncate here since vector.DynamicBuilder.Write doesn't retain a reference to the bytes returned by val.Bytes().
```suggestion
		b.Truncate()
```